src/main/java/se/iths/librarysystem/controller/BookController.java

```diff
@@ -48,4 +49,19 @@ public BookController(BookService bookService, ModelMapper modelMapper) {
                 .body(savedBook);
     }
 
+    @DeleteMapping("{id}")
+    public ResponseEntity<Void> deleteBook(@PathVariable Long id){
+
+        bookService.deleteBook(id);
+        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+    }
+
+    @GetMapping("{id}")
+    public ResponseEntity<Optional<BookEntity>> findBookByID(@PathVariable Long id) {
+        Optional<BookEntity> foundBook = bookService.findBookById(id);
+        return new ResponseEntity<>(foundBook, HttpStatus.OK);
+
+    }
```

Comment:
```suggestion
    public ResponseEntity<Book> findBookByID(@PathVariable Long id) {
        BookEntity foundBook = bookService.findBookById(id).orElseThrow(() -> new IdNotFoundException("books", id));
        // convert BookEntity to a Book DTO
        return new ResponseEntity<>(bookDTO, HttpStatus.OK);

    }
```

I suggest that:
- we add exception handling if the book id does not exist. 
- the method returns a Book DTO according to the project's requirements below:
> _Enbart modeller ska returneras till användaren._